src/python/espressomd/interactions.pyx

```diff
@@ -2250,6 +2250,11 @@ class ThermalizedBond(BondedInteraction):
         distance vector of the particle pair.
     r_cut: :obj:`float`, optional
         Specifies maximum distance beyond which the bond is considered broken.
+    seed : :obj:`int`
+        Counter value of the philox RNG. Initially, this defines the seed of
+        the RNG. If prompted, it does not return the initially set counter value
+        (the seed) but the current state of the RNG.
```

Comment:
1. If a seed is provided on creation of the bond, the global counter for all TBs is set to that value. I realise now that this causes trouble in the (very special) case of a second (third..) TB with the same seed that will 'reset' the thermalization of all the TBs.
2. Thanks!

Reply:
What do you think of this approach:
```python
    def _set_params_in_es_core(self):
        if thermalized_bond_is_seed_required():
            if self.params["seed"] is None:
                raise ValueError(
                    "A seed has to be given as keyword argument on first activation of the thermalized bond")
            else:
                utils.check_type_or_throw_except(
                    self.params["seed"], 1, int, "seed must be a positive integer")
                thermalized_bond_set_rng_state(self.params["seed"])
        elif self.params["seed"] is not None:
            raise ValueError(
                "A seed has to be given as keyword argument only on first activation of the thermalized bond")

        thermalized_bond_set_params(
            self._bond_id, self._params["temp_com"], self._params["gamma_com"],
            self._params["temp_distance"], self._params["gamma_distance"], self._params["r_cut"])
```

```python
>>> import espressomd
>>> system = espressomd.system.System(box_l=3*[12])
>>> tb1 = espressomd.interactions.ThermalizedBond(temp_com=300,gamma_com=50,temp_distance=300,gamma_distance=20,r_cut=1)
>>> tb2 = espressomd.interactions.ThermalizedBond(temp_com=300,gamma_com=50,temp_distance=300,gamma_distance=20,r_cut=1,seed=5)
>>> system.bonded_inter.add(tb1) # fails: has no seed
ValueError: A seed has to be given as keyword argument on first activation of the thermalized bond
>>> system.bonded_inter.add(tb2) # ok: has a seed
>>> system.bonded_inter.add(tb2) # fails: overwrites seed
ValueError: A seed has to be given as keyword argument only on first activation of the thermalized bond
>>> system.bonded_inter.add(tb1) # ok: has no seed
```

Reply:
This approach can be problematic when setting TBs in a for loop. Also when setting a TB in multiple test methods of a unittest class, since the order of the tests is based on the alphabetical order of the methods and the presence/absence of fixtures, the only way to reliably set the seed only once is to implement an MPI-safe lock mechanism... We'll just leave the seed logic as it currently is.